### PR TITLE
introduce pyright + fix issues (supersedes #1356)

### PR DESCRIPTION
**Pull request by prusnak**
Rework of https://github.com/lnbits/lnbits/pull/1356 (by @dni) to not touch files that do not need touching

Functionally these are very much the same changes as in #1356

### CODE REVIEW

**Comment by arcbtc** [comment]
This is the last big blanket change before we pull out all extensions. How is health of this pr, will it impact functionality, what needs to be tested?

**Comment by prusnak** [comment]
> What is health of this pr, will it impact functionality, what needs to be tested?

I just rebased https://github.com/lnbits/lnbits/pull/1356 - overall the changes are usually simple refactors with added assert statements. Maybe @dni can answer the question whether something needs to be tested better?

File: lnbits/wallets/lnpay.py [hunk 1]
```diff
@@ -133,27 +133,27 @@ async def paid_invoices_stream(self) -> AsyncGenerator[str, None]:
...
-        async with httpx.AsyncClient() as client:
-            r = await client.get(
-                f"{self.endpoint}/lntx/{lntx_id}?fields=settled", headers=self.auth
-            )
-            data = r.json()
-            if data["settled"]:
-                await self.queue.put(lntx_id)
-
-        raise HTTPException(status_code=HTTPStatus.NO_CONTENT)
+    # async def webhook_listener(self):
```
Comment (dni) [inline lnbits/wallets/lnpay.py:136]: this in unintended

File: lnbits/wallets/opennode.py [hunk 2]
```diff
@@ -139,17 +142,17 @@ async def paid_invoices_stream(self) -> AsyncGenerator[str, None]:
             value = await self.queue.get()
             yield value
 
-    async def webhook_listener(self):
-        data = await request.form
-        if "status" not in data or data["status"] != "paid":
-            raise HTTPException(status_code=HTTPStatus.NO_CONTENT)
+    # async def webhook_listener(self):
```
Comment (dni) [inline lnbits/wallets/opennode.py:145]: this is unintended

**Comment by dni** [comment]
@prusnak  can you make me contributor to your fork ;D? i want to push there

**Comment by prusnak** [comment]
Superseded by https://github.com/lnbits/lnbits/pull/1468

**Pull request closed without merging**